src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,65 @@
+package parser;
+
+import command.AddCommand;
+import command.FindCommand;
+import command.ExitCommand;
+import command.DeleteCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.UnmarkCommand;
+import command.Command;
+import exceptions.BuddyException;
+
+/**
+ * Parses user input and returns the appropriate command.
+ */
+public class Parser {
+
+    /**
+     * Parses the given command string and returns a corresponding Command object.
+     *
+     * @param command The user input command string.
+     * @return The Command object corresponding to the user input.
+     * @throws BuddyException If the command is not recognized or is invalid.
+     */
+    public static Command parse(String command) throws BuddyException {
+        String[] commandParts = command.split(" ");
+        String mainCommand = commandParts[0].toLowerCase();
+
+        switch (mainCommand) {
+            case "todo":
+            case "deadline":
+            case "event":
+                return new AddCommand(command);
+            case "delete":
+                return new DeleteCommand(parseTaskIndex(commandParts));
+            case "mark":
+                return new MarkCommand(parseTaskIndex(commandParts));
+            case "unmark":
+                return new UnmarkCommand(parseTaskIndex(commandParts));
+            case "bye":
+                return new ExitCommand();
+            case "list":
+                return new ListCommand();
+            case "find":
+                return new FindCommand(commandParts[1].trim());
+            default:
```

Comment:
Should the case and default lines be in-line with the Switch statement as according to the coding standards?